Q2.cpp

```diff
@@ -0,0 +1,83 @@
+//CodeU: Q2
+//
+//  Created by Julia Semavina on 18.05.2017
+//
+//  problem specification:
+//      Implement an algorithm to find the kth to last element of a singly linked list
+//
+//  comments:
+//      list length is unknown
+//      if K is bigger than the size of the list the answer will be IMPOSSIBLE INPUT
+//      KthToLast function contains a solution
+//
+
+#include <iostream>
+#include <vector>
+
+
+struct Node {
+    char value;
+    Node* next;
+};
+
+Node* CreateSinglyLinkedList() {
+    Node *Head = new Node;
+    std::cin >> Head->value;
+    Node *elem1 = Head;
+    Node *elem2 = new Node;
+    while (std::cin >> elem2->value) {
+        elem1->next = elem2;
+        elem1 = elem2;
+        elem2 = new Node;
+    }
+    elem1->next = NULL;
+    return (Head);
+};
+
+Node* KthToLast(Node* Head, int K, bool &problem) {
```

Comment:
No need for a bool parameter, you can return NULL when the input is invalid.
Also, reference parameters are usually discouraged, because it's not clear when you call the function that the parameter may be modified, we usually use pointers for that.
Compare:
Node* Result = KthToLast(Head, K, problem);
With
Node* Result = KthToLast(Head, K, &problem);
In the second is very clear that 'problem' may be modified.